packages/babel-parser/src/plugins/typescript/scope.ts

```diff
@@ -35,11 +38,50 @@ class TypeScriptScope extends Scope {
 // explanation of how typescript handles scope.
 
 export default class TypeScriptScopeHandler extends ScopeHandler<TypeScriptScope> {
+  importsStack: Set<string>[] = [];
+
   createScope(flags: ScopeFlags): TypeScriptScope {
+    this.importsStack.push(new Set()); // SCOPE_TS_TOP_LEVEL should be kept
```

Comment:
Did you mean `SCOPE_TS_MODULE`?

Reply:
https://github.com/babel/babel/blob/14ea25fbb337f7c7cc6c29ae5ee480fc76818087/packages/babel-parser/src/plugins/typescript/index.ts#L1861-L1875
There seems to be an extra `SCOPE_OTHER` here.
At first I changed it to `SCOPE_TS_TOP_LEVEL`, but then I found it was not necessary.
Because export detection doesn't happen until all scopes have exited, I must and should always keep the top-level scope.
Perhaps remove the comment or modify it to:
Always keep the top-level scope for export checks.